src/differential/mod.rs

```diff
@@ -0,0 +1,186 @@
+use hashbrown::HashMap;
+use std::fs::File;
+use std::io;
+use std::io::prelude::*;
+use std::path::Path;
+
+const READER_CAPACITY: usize = 128 * 1024;
+
+#[derive(Debug, Clone, Copy, Default)]
+struct Counts {
+    first: usize,
+    second: usize,
+}
+
+/// Configure the generated output.
+///
+/// All options default to off.
+#[derive(Debug, Clone, Copy, Default)]
+pub struct Options {
+    /// Normalize the first profile count to match the second.
+    ///
+    /// This can help in scenarios where you take profiles at different times, under varying
+    /// load. If you generate a differential flame graph without setting this flag, everything
+    /// will look red if the load increased, or blue if it decreased. If this flag is set,
+    /// the first profile is balanced so you get the full red/blue spectrum.
+    pub normalize: bool,
+
+    /// Strip hex numbers (addresses) of the form "0x45ef2173" and replace with "0x...".
+    pub strip_hex: bool,
+}
+
+/// Produce an output that can be used to generate a differential flame graph.
+///
+/// The readers are expected to contain folded stack lines of before and after profiles with
+/// the following whitespace-separated fields:
+///
+///  - A semicolon-separated list of frame names (e.g., `main;foo;bar;baz`).
+///  - A sample count for the given stack.
+///
+/// The output written to the `writer` will be similar to the inputs, except there will be two
+/// sample count columns -- one for each profile.
+pub fn from_readers<R1, R2, W>(opt: Options, reader1: R1, reader2: R2, writer: W) -> io::Result<()>
```

Comment:
I wonder if we should give `reader1` and `reader2` better names, or at least mention those names explicitly in the text above, so the reader knows which one is "before" and which one is "after".

Reply:
Should we just name the `before` and `after`?